bin/generate.py

```diff
@@ -2002,11 +2003,155 @@ def get_or_add(yaml, key, t=ruamel.yaml.comments.CommentedMap):
 
         write_yaml(data, generators_yaml)
         bar.finalize()
-        return
+        return True
 
+    # reorder all testcases in the given directories
+    def reorder(self):
+        if self.n_parse_error > 0:
+            return False
 
-# Delete files in the tmpdir trash directory. By default all files older than 10min are removed
-# and additionally the oldest files are removed until the trash is less than 1 GiB
+        if not has_ryaml:
+            error(
+                'generate --reorder needs the ruamel.yaml python3 library. Install python[3]-ruamel.yaml.'
+            )
+            return False
+
+        directory_rules = set()
+        for d in config.args.testcases:
+            path = d.relative_to('data')
+            parts = path.parts
+            if not parts:
+                warn(f'Cannot reorder Root directory. Skipping.')
+            elif parts[0] in config.INVALID_CASE_DIRECTORIES:
+                warn(f'{d} is used for invalid test data. Skipping.')
+            elif path not in self.known_directories:
+                warn(f'{d} is not a generated directory. Skipping.')
+            elif not self.known_directories[path].numbered:
+                warn(f'{d} is not numbered. Skipping.')
+            elif not self.known_directories[path].data:
+                warn(f'{d} is empty. Skipping.')
+            else:
+                directory_rules.add(self.known_directories[path])
+
+        data = self.problem.path / 'data'
+
+        testcase_filter = set()
+        for d in directory_rules:
+            for c in d.data:
+                testcase_filter.add(data / c.path.with_suffix('.in'))
+
+        ts_pair = self.problem.prepare_run()
+        if ts_pair == False:
+            return False
+
+        testcases = [t for t in ts_pair[0] if t.in_path in testcase_filter]
+        submissions = [s for s in ts_pair[1] if s.expected_verdicts != [Verdict.ACCEPTED]]
+
+        if not testcases:
+            error(f'No testcases found.')
+            return False
+        if not submissions:
+            error(f'No rejected submissions found.')
+            return False
+
+        ok, verdict_table = Problem.run_some(testcases, submissions)
```

Comment:
This `ok` can be `False` when running WA submissions on only a subset (subdir) of cases. Probably we should just ignore it.